bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/io/RefreshingOAuth2CredentialsInterceptor.java

```diff
@@ -325,8 +326,13 @@ HeaderCacheElement syncRefresh() {
         this.futureToken = executor.submit(new Callable<HeaderCacheElement>() {
           @Override
           public HeaderCacheElement call() throws Exception {
-            HeaderCacheElement newToken = refreshCredentials();
-            return updateToken(newToken);
+            try {
+              HeaderCacheElement newToken = refreshCredentials();
+              return updateToken(newToken);
+            } finally {
+              futureToken = null;
+              isRefreshing = false;
```

Comment:
How come this moved from updateToken? I think it needs to be part of that synchronization block

Reply:
Sorry about that. I reverted that change.